torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -2215,10 +2215,39 @@ struct to_ir {
         if (stmt.type().present()) {
           type = typeParser_.parseTypeFromExpr(stmt.type().get());
         }
+        auto rhs_sugared_val = emitSugaredExpr(rhs, 1, type);
+        // START BC HACK
+        //
+        // For old serialized quantized RNN modules, switch quantized::linear_prepack
+        // to quantized::linear_prepack_legacy. We changed linear_prepack to return
+        // a TorchBind class and not a cpp_custom_type_hack tensor anymore, but the
+        // old serialized models are tightly coupled with the type_hack version.
+        // If we still create a Tensor here, then the quantized_lstm.legacy overload
+        // can kick in in forward_impl(), and the module will still run correctly.
```

Comment:
Under what situation can this BC hack be removed?

Reply:
Either of:
- We don't see any production models with this serialized pattern anymore, or
- We make a call that we're no longer supporting that old version of the quantized RNN module in OSS